src/services/EcoBridge/OmniBridge/OmniBridge.utils.ts

```diff
@@ -0,0 +1,824 @@
+
+  if (!enableReversedBridge) {
+    return token.chainId === homeChainId ? 'erc677' : 'erc20'
+  }
+
+  const { chainId, address } = token
+  const mediatorAddress = getMediatorAddressWithoutOverride(bridgeDirection, chainId)
+
+  const abi = ['function nativeTokenAddress(address) view returns (address)']
+
+  if (!mediatorAddress) return
+
+  const mediatorContract = new Contract(mediatorAddress, abi, provider)
+  const nativeTokenAddress = await mediatorContract.nativeTokenAddress(address)
+
+  if (nativeTokenAddress === ADDRESS_ZERO) return 'erc20'
+
+  return 'erc677'
+}
+
+//get name
+export const fetchTokenName = async (token: TokenWithAddressAndChain & { name: string }, provider?: Provider) => {
+  let tokenName = token.name || ''
+
+  try {
+    const stringAbi = ['function name() view returns (string)']
+    const tokenContractString = new Contract(token.address, stringAbi, provider)
+    tokenName = await tokenContractString.name()
+  } catch {
+    const bytes32Abi = ['function name() view returns (bytes32)']
+    const tokenContractBytes32 = new Contract(token.address, bytes32Abi, provider)
+    tokenName = utils.parseBytes32String(await tokenContractBytes32.name())
+  }
+  return tokenName
+}
+
+const getToName = async (
+  fromToken: TokenWithAddressAndChain & { name: string },
+  toChainId: ChainId,
+  toAddress: string,
+  provider?: Provider
+) => {
+  const { name } = fromToken
+  if (toAddress === ADDRESS_ZERO) {
+    const fromName = name || (await fetchTokenName(fromToken))
+    return fromName
+  }
+
+  return fetchTokenName({ chainId: toChainId, address: toAddress, name }, provider)
+}
+
+//get mediator
+export const getMediatorAddress = (bridgeDirection: string, token: TokenWithAddressAndChain) => {
+  if (!token || !token.chainId || !token.address) return
+
+  if (isOverridden(bridgeDirection, token)) {
+    return getOverriddenMediator(bridgeDirection, token)
+  }
+
+  return getMediatorAddressWithoutOverride(bridgeDirection, token.chainId)
+}
+
+//fetch toToken details
+const fetchTokenDetailsString = async (token: TokenWithAddressAndChain, provider?: Provider) => {
+  const abi = [
+    'function decimals() view returns (uint8)',
+    'function symbol() view returns (string)',
+    'function name() view returns (string)',
+  ]
+  const tokenContract = new Contract(token.address, abi, provider)
+
+  const [name, symbol, decimals] = await Promise.all<string, string, number>([
+    tokenContract.name(),
+    tokenContract.symbol(),
+    tokenContract.decimals(),
+  ])
+
+  return { name, symbol, decimals }
+}
+
+const fetchTokenDetailsBytes32 = async (token: TokenWithAddressAndChain, provider?: Provider) => {
+  const abi = [
+    'function decimals() view returns (uint8)',
+    'function symbol() view returns (bytes32)',
+    'function name() view returns (bytes32)',
+  ]
+  const tokenContract = new Contract(token.address, abi, provider)
+  const [name, symbol, decimals] = await Promise.all<string, string, number>([
```

Comment:
Tuple will work here

```suggestion
  const [name, symbol, decimals] = await Promise.all<[string, string, number]>([
```